src/main/java/com/jmb/springfactory/service/task/TaskServiceImpl.java

```diff
@@ -29,122 +34,185 @@
-   * @param orderId
-   * @return
-   * @throws PersistenceLayerException
-   */
-  private Integer getLastPositionOrder(Integer orderId) throws PersistenceLayerException {
-
-    final Integer numberOrders = taskMySQLService.countByOrderId(orderId).intValue();
-    final Integer lastNumberOrder = numberOrders + 1;
-
-    serviceLog.info(String.format("There are %s task for order %s, the order of the new one is %s", 
-        orderId, numberOrders, lastNumberOrder));
-
-    return lastNumberOrder;
-  }
-
-  @Override
-  public List<TaskDto> findAll(QueryTaskObject queryParams) {
-    serviceLog.info(String.format("Search task with the follow arguments: %s", queryParams.toString()));
-    return taskMySQLService.findAll(queryParams).map(this::entityToDto).collect(Collectors.toList());
-  }
-  
-  @SuppressWarnings("unchecked")
-  @Override
-  public void addComment(Integer idTask, CommentDto comment) throws NotFoundException, ServiceLayerException {
-    
-    final TaskDto task = findOne(idTask);
-    task.setComments((List<CommentDto>)addIntoList(task.getComments(), comment));
-    save(task);
-  }
-
-  @SuppressWarnings("unchecked")
-  @Override
-  public void addWorkLog(Integer idTask, WorkLogDto workLog) throws NotFoundException, ServiceLayerException {
-    
-    final TaskDto task = findOne(idTask);
-    task.setWorkLogs((List<WorkLogDto>)addIntoList(task.getWorkLogs(), workLog));
-    save(task);
-  }
-  
-  @Override
-  public Optional<TaskDto> findOneById(Integer id) {
-    return taskMySQLService.findOne(id).map(this::entityToDto);
-  }
-  
+public class TaskServiceImpl extends GenericServiceImpl<Task, TaskDto, BusinessObjectBase, Integer>
+        implements TaskService {
+
+    @Autowired
+    private TaskMySQLService taskMySQLService;
+
+    @Autowired
+    private ProductionOrderService productionOrderService;
+
+    @Autowired
+    private GroupService groupService;
+
+    @Autowired
+    @Qualifier("taskValidatorService")
+    private ValidatorService taskValidatorService;
+
+    private static final String TASK_INITIAL_STATUS = TaskStatusEnum.OPENED.name();
+
+    @Override
+    public GenericMySQLService<Task, Integer> genericDao() {
+        return taskMySQLService;
+    }
+
+    @Override
+    public Class<? extends Task> getClazz() {
+        return Task.class;
+    }
+
+    @Override
+    public Class<? extends TaskDto> getDtoClazz() {
+        return TaskDto.class;
+    }
+
+    @Override
+    public Class<? extends BusinessObjectBase> getBoClazz() {
+        return BusinessObjectBase.class;
+    }
+
+    @Override
+    public void update(TaskDto taskDto, Integer idTask) throws ServiceLayerException {
+        taskValidatorService.validateOnUpdate(taskDto);
+        super.update(taskDto, idTask);
+    }
+
+    @Override
+    public TaskDto save(Integer orderId, TaskDto taskDto)
+            throws ServiceLayerException, NotFoundException, PersistenceLayerException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)